bob-backend/src/pearl/group.rs

```diff
@@ -534,7 +534,9 @@ impl Group {
             {
                 if let Some(name) = PartitionName::try_from_string(&file_name) {
                     let pearl_holder = self.create_pearl_holder(name.timestamp, &name.hash);
-                    holders.push(pearl_holder);
+                    let pos = holders.binary_search_by(
+                        |holder: &Holder| holder.start_timestamp().cmp(&name.timestamp));
```

Comment:
For safety, use `holder.start_timestamp()` instead of name.timestamp.
Also, `sort_by` at the end would be simpler (just 1 line)